pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1717,23 +1717,62 @@ private void updateRowBuffer() throws SQLException {
           //
 
           case Types.DATE:
-            rowBuffer[columnIndex] = connection
-                .encodeString(
-                    connection.getTimestampUtils().toString(
-                        getDefaultCalendar(), (Date) valueObject));
+            if (valueObject == null || valueObject instanceof Timestamp) {
```

Comment:
I don't think any of these three `null` checks work. I believe if `null` is encountered a `NullPointerException` will happen in `TimestampUtils` because unconditionally `#getTime()` is called.

Reply:
@marschall You are correct, it valueobject isn't null here even if you insert/update with a null value.

I have added unit tests that displays this and removed the three checks.